corehq/apps/reports/standard/deployments.py

```diff
@@ -209,8 +210,33 @@ def user_query(self, pagination=True):
                                        )
         return user_query
 
+    def get_location_columns(self, grouped_ancestor_locs):
+        from corehq.apps.locations.models import LocationType
+        location_types = LocationType.objects.by_domain('icds-cas')
+        all_user_locations = grouped_ancestor_locs.values()
+        all_user_loc_types = {loc.location_type_id for user_locs in all_user_locations for loc in user_locs}
+        required_loc_columns = [loc_type for loc_type in location_types if loc_type.id in all_user_loc_types]
+        return required_loc_columns
+
+    def user_locations(self, ancestors, location_type):
+        location_list = ['---'] * len(location_type)
+        types_id = [type.id for type in location_type]
```

Comment:
nit: it should be `location_types` and `type_ids`

Reply:
nit: its probably best not to use a builtin for the iteration variable